lib/pbench/test/unit/server/query_apis/test_datasets_mappings.py

```diff
@@ -1,20 +1,5 @@
 from http import HTTPStatus
 
-import pytest
-from sqlalchemy.exc import DatabaseError
-
-from pbench.server.database.models.template import Template
-
-
-@pytest.fixture()
-def database_error(monkeypatch):
-    def raise_db_error(name: str):
-        raise DatabaseError("DB Error")
-
-    with monkeypatch.context() as m:
-        m.setattr(Template, "find", raise_db_error)
-        yield
```

Comment:
Why isn't this mock needed anymore?

Reply:
It is because if the `find` can not talk to the database its anyway gonna result in some sort of sqlalchemy database error and we catch it with broad `SQLAlchemyError` so we don't have to raise it explicitly

Reply:
So...we're counting on the environment, by default, aligning so that the database happens to be inaccessible?

Wouldn't it be better to establish the behavior that we want the CUT to encounter?  That way, there is no question what `find()` should see during the test, and therefore no question how the CUT should behave.

Reply:
Yeah I am little confused now tbh, my thinking behind this was to make sure we catch any exception that can be raised (since we are catching the broad `SQLAlchemyError` exception) and, by default, its raising the database error so no need for this fixture. But you have a point, this fixture would establish the fixed behavior that our CUT will encounter.

Reply:
> I am little confused now tbh

My bad.  I'm sorry about that.

Where we should be (based on the changes you've made to the CUT) is that we should have a unit test for `DatasetsMappings._get()` which uses a mocked `IndexMapBase.get_mappings()`, which does or does not (depending on the scenario) raise `TemplateNotFound`.  And, separately, we should have a unit test for `IndexMapBase.get_mappings()`, which uses a mocked `Template.find()` to drive its behaviors.  (And, somewhere in there, we should probably have an assertion that checks that the stringization of `TemplateNotFound` is satisfactory.)

However, this part of the code has typically been tested more functionally, rather than in discrete units.  So, I'm guessing that that is where our respective mindsets may not be fitting together very well.

Previous to your change, it looks like there were two tests for the `/datasets/mappings/summary` URI, one with a `find()` mock which returns lots of data, and one with a `find()` mock which raised an exception.  Your change removes the latter mock, depending instead on the exception being raised more or less by default or coincidence.

Your assertion that we don't need this latter mock is technically correct, but it leads to a fragile and somewhat confusing test regimen.  So, we really should have this mock, so that we can control the behavior that the CUT encounters, so that we can have confidence that the test produces the appropriate result.